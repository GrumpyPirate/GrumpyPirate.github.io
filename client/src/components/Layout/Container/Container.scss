@import 'core';

.container {
  width: 100%;
  padding-right: rem(map-get($gutter-widths, xs) / 2);
  padding-left: rem(map-get($gutter-widths, xs) / 2);

  @each $breakpoint in map-keys($gutter-widths) {
    @if index($breakpoints, $breakpoint map-get($breakpoints, $breakpoint)) > 1 {
      @include respond-up($breakpoint) {
        padding-right: rem(map-get($gutter-widths, $breakpoint) / 2);
        padding-left: rem(map-get($gutter-widths, $breakpoint) / 2);
      }
    }
  }

  // Modifiers
  &--is {
    &-fluid {
      width: 100%;
    }
  }
}
