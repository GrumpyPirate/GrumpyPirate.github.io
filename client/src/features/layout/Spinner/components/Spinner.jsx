import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Icon from 'components/Icon/Icon';

import './Spinner.scss';

const Spinner = ({ isLoading }) => (
  <div
    className={classnames('spinner', {
      'spinner--is-loading': isLoading,
    })}
  >
    <figure className="spinner__icon">
      <Icon glyph="loader" altText="Loading..." />
    </figure>

    <h6 className="h4 mb-0">Loading...</h6>
  </div>
);

Spinner.propTypes = {
  isLoading: PropTypes.bool.isRequired,
};

export default Spinner;
