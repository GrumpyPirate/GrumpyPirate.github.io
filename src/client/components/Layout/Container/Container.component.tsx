import React, { FunctionComponent } from 'react';

import { ChildrenProps } from 'types';

import classes from './Container.component.scss';

const Container: FunctionComponent<ChildrenProps> = ({ children }: ChildrenProps) => (
  <div className={classes['container']}>{children}</div>
);

export default Container;
