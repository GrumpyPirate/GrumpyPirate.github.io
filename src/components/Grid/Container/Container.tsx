import styled from 'styled-components';

import { gutterWidths, mediaQueries, rem } from 'styles';

const Container = styled.div`
  width: 100%;
  padding-right: ${rem(gutterWidths.xs / 2)};
  padding-left: ${rem(gutterWidths.xs / 2)};

  @media ${mediaQueries.md} {
    padding-right: ${rem(gutterWidths.md / 2)};
    padding-left: ${rem(gutterWidths.md / 2)};
  }
`;

export default Container;
