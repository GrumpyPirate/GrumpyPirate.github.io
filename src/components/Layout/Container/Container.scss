@import 'core';

.container {
  padding-left: rem(map-get($gutter-widths, 'xs') / 2);
  padding-right: rem(map-get($gutter-widths, 'xs') / 2);
  width: 100%;

  @each $breakpoint in map-keys($gutter-widths) {
    @if index($breakpoints, $breakpoint map-get($breakpoints, $breakpoint)) > 1 {
      @include respond-up($breakpoint) {
        padding-left: rem(map-get($gutter-widths, $breakpoint) / 2);
        padding-right: rem(map-get($gutter-widths, $breakpoint) / 2);
      }
    }
  }

  // Modifiers
  &--is {
    &-fluid {
      width: 100%;
    }
  }
}
