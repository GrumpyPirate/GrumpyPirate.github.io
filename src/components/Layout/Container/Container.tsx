import React, { FunctionComponent } from 'react';

import { ChildrenProps } from 'types/common';

import classes from './Container.scss';

const Container: FunctionComponent<ChildrenProps> = ({ children }) => (
  <div className={classes['container']}>{children}</div>
);

export default Container;
