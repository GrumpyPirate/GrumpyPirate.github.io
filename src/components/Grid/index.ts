import Column from './Column/Column';
import Container from './Container/Container';
import Row from './Row/Row';

export { Container, Row, Column };
