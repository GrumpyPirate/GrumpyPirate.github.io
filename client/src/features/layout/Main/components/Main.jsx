import React, { PureComponent, createRef } from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

import Spinner from '../../Spinner/containers/Spinner';

import './Main.scss';

class Main extends PureComponent {
  constructor(props) {
    super(props);

    this.mainElement = createRef();
  }

  componentDidUpdate(prevProps) {
    const { location: prevLocation } = prevProps;
    const { location } = this.props;

    if (prevLocation.pathname !== location.pathname) {
      this.mainElement.current.scrollTo(0, 0);
    }
  }

  render() {
    const { isLoading, children } = this.props;

    return (
      <main
        className={classnames('main', {
          'is--loading': isLoading,
        })}
        ref={this.mainElement}
      >
        <Spinner />
        {children}
      </main>
    );
  }
}

Main.propTypes = {
  isLoading: PropTypes.bool,
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
  }).isRequired,
  children: PropTypes.node.isRequired,
};

Main.defaultProps = {
  isLoading: false,
};

export default Main;
