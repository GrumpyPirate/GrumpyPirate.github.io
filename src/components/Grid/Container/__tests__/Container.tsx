import { render } from '@testing-library/react';
import React from 'react';

import { Column, Row } from 'components/Grid';

import Container from '../Container';

describe('Components/Grid/Container', () => {
  const props = {
    children: (
      <>
        <Row>
          <Column>Column 1</Column>
          <Column>Column 2</Column>
          <Column>Column 3</Column>
        </Row>
      </>
    ),
  };

  test('should render as expected, without crashing', () => {
    const { container } = render(<Container {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
