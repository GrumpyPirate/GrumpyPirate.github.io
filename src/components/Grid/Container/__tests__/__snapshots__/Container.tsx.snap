// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Grid/Container should render as expected, without crashing 1`] = `
.c2 {
  position: relative;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  max-width: 100%;
  min-height: 1px;
  padding-right: 0.75rem;
  padding-left: 0.75rem;
}

.c0 {
  width: 100%;
  padding-right: 0.75rem;
  padding-left: 0.75rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  margin-right: -0.75rem;
  margin-left: -0.75rem;
}

@media (min-width:48em) {
  .c2 {
    padding-right: 0.9375rem;
    padding-left: 0.9375rem;
  }
}

@media (min-width:48em) {
  .c0 {
    padding-right: 0.9375rem;
    padding-left: 0.9375rem;
  }
}

@media (min-width:48em) {
  .c1 {
    margin-right: -0.9375rem;
    margin-left: -0.9375rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      Column 1
    </div>
    <div
      class="c2"
    >
      Column 2
    </div>
    <div
      class="c2"
    >
      Column 3
    </div>
  </div>
</div>
`;
